runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpStateSingle.java

```diff
@@ -117,6 +132,60 @@ private int getActionsRemaining()
 		return Integer.MAX_VALUE;
 	}
 
+	private int getKillsRemaining(Client client)
+	{
+		Player local = client.getLocalPlayer();
+		if (local == null)
+		{
+			return Integer.MAX_VALUE;
+		}
+		Actor opponent = local.getInteracting();
```

Comment:
Once again this would only work during combat and as soon as you're out of combat it would stop working.

Reply:
This is a valid point. Also the `XPStateSingle` should not be responsible for calculating something like this in real time. This logic should be within the xptracker plugin. Maybe the best event to use for this would be `NpcLootReceived` since you can kill things you are not interacting with (cannon?). Get the health of the NPC which died and calculate it into the running average.

Reply:
I am not too comfortable with so much logic here. This is meant to be a domain object that stores mostly data.